### DISCUSSION TITLE
Potential SPARQL Queries

### DISCUSSION THREAD

**Discussion by tom-james-watson** [tom-james-watson/wikitrivia-scraper#8] [General]
In order to improve the complexity of cards, make card generation simpler and to allow for the creation of card collections, we want to move to using SPARQL queries to generate the game cards. This discussion is to serve as a place for people to share interesting SPARQL queries upon which we could potentially base the card generation. See issue https://github.com/tom-james-watson/wikitrivia-scraper/issues/6 for some background information.

Here is what I think we would need in the query results:

- id
- label
- date prop id
- date
- description
- wikipedia title
- instance of
- sitelinks count

**Comment by namedgraph** [comment]
I think you should identify properties such as `wdt:P577` (publishing date) that can give you useful dates, and from there you could find the most popular instances using `GROUP BY ?count`/`ORDER BY DESC (?count)`.

**Comment by tuukka** [comment]
Here's a very generic query to get us started (for simplicity, hardcoding inception as the date property): https://qlever.cs.uni-freiburg.de/wikidata/uBQWSn

It was working before I added the part about class/instance of, now I may have crashed the service :fearful:

**Reply by tom-james-watson** [reply]
This seems a LOT more performant than the official wikidata API! Thanks for the link :+1:

**Reply by tuukka** [reply]
Here's a more complete query from tom-james-watson/wikitrivia#26: https://qlever.cs.uni-freiburg.de/wikidata/OycBUK

**Reply by tom-james-watson** [reply]
EDIT - I'm going to move the discussion to https://github.com/tom-james-watson/wikitrivia-generator/issues/6.
